### DISCUSSION TITLE
How can I set default UN/PW of xrv9k image

### DISCUSSION THREAD

**Discussion by ACodingfreak** [srl-labs/containerlab#1830] [Q&A]
Hi All,

Currently when I create a new container image for xrv9k VM image, by default clab/clab@123 is the default username and password for ssh login. Where can I reset the same in vrnetlab package ?

When I used standard vrnetlab repo, I was changing the same in launch.py. But this does not seems to work with containerlab.
Any pointers would be of great help

**Comment by hellt** [comment]
Hi, you can override the default value here https://github.com/hellt/vrnetlab/blob/302a2075b2d2a72f0e83c91a1d583aa8e63c7799/xrv9k/docker/launch.py#L340

Originally it is set in containerlab and passed to vrnetlab launcher.py

**Reply by ACodingfreak** [reply]
@hellt 

I did try the changes suggested by you, but it did not work.

When I create a new xrv9k container using vrnetlab I am seeing default UN/PW as clab/clab@123. As you can see in launch.py, default UN/PW is vrnetlab/VR-netlab9 instead of clab/clab@123. So I am not sure currently where are we overriding the default UN/PW with clab/clab@123

**Reply by hellt** [reply]
make sure you're looking at `hellt/vrnetlab` repo and not the `vrnetlab/vrnetlab`

**Reply by ACodingfreak** [reply]
Well I am using hellt/vrnetlab repo which you have mentioned 

```
ubuntu@op2:~/containerlab/test$ git clone https://github.com/hellt/vrnetlab.git
Cloning into 'vrnetlab'...
remote: Enumerating objects: 4288, done.
remote: Counting objects: 100% (1283/1283), done.
remote: Compressing objects: 100% (361/361), done.
remote: Total 4288 (delta 1030), reused 1061 (delta 922), pack-reused 3005
Receiving objects: 100% (4288/4288), 1.99 MiB | 1.38 MiB/s, done.
Resolving deltas: 100% (2595/2595), done.
ubuntu@op2:~/containerlab/test$ 
ubuntu@op2:~/containerlab/test$ 
ubuntu@op2:~/containerlab/test$ ls
vrnetlab
ubuntu@op2:~/containerlab/test$ cd vrnetlab/
ubuntu@op2:~/containerlab/test/vrnetlab$ ls
CODE_OF_CONDUCT.md  Makefile   asav    ftdv             makefile-install.include  n9kv   openbsd  routeros  vjunosevolved  vqfx         vsr1000  xrv9k
CONTRIBUTING.md     README.md  common  ftosv            makefile-sanity.include   nxos   openwrt  sros      vjunosswitch   vrnetlab.sh  vsrx
LICENSE             aoscx      csr     git-lfs-repo.sh  makefile.include          ocnos  pan      veos      vmx            vrp          xrv
ubuntu@op2:~/containerlab/test/vrnetlab$ cd xrv9k/docker/
ubuntu@op2:~/containerlab/test/vrnetlab/xrv9k/docker$ 

ubuntu@op2:~/containerlab/test/vrnetlab/xrv9k/docker$ cat launch.py | grep -rin "parser.add_argument"
launch.py:307:    parser.add_argument(
launch.py:310:    parser.add_argument("--hostname", default="vr-xrv9k", help="Router hostname")
launch.py:311:    parser.add_argument("--username", default="vrnetlab", help="Username")
launch.py:312:    parser.add_argument("--password", default="VR-netlab9", help="Password")
launch.py:313:    parser.add_argument("--nics", type=int, default=128, help="Number of NICS")
launch.py:314:    parser.add_argument(
launch.py:317:    parser.add_argument(
launch.py:320:    parser.add_argument(
ubuntu@op2:~/containerlab/test/vrnetlab/xrv9k/docker$ 

```

**Reply by hellt** [reply]
as I mentioned, you can override the value in the highlighted line
https://github.com/hellt/vrnetlab/blob/302a2075b2d2a72f0e83c91a1d583aa8e63c7799/xrv9k/docker/launch.py#L340

**Reply by ACodingfreak** [reply]
I will try out hardcoding the username/password in designated line.

Just curious.

When you said "Originally it is set in containerlab and passed to vrnetlab launcher.py". What will be the command containerlab will call to pass this information to launcher.py.

**Reply by hellt** [reply]
Containerlab passes username and password to the launch.py as command arguments
